makepdf.sh

```diff
@@ -87,6 +110,19 @@ get_file_path()
         echo $(dirname "$infile")
 }
 
+get_out_rel_path_file()
+{
+        input_basename="$1"
+        format="$2"
+
+        out_rel_path_file=""
+
+        out_rel_path_file="./"$format"/"$input_basename"."$format
```

Comment:
What about epub? 

This PR fixes the generation of epub (separate epub path, -d, I think, was also broken for pdf and epub); more importantly epub (and pdf) does require the logo. These markdown systems all have a kind of if (pdf || epbub) {wherever/path/to/logo} else {somewhere/else/logo} which  I didn't really investigate in mkdocs. We can also specify the logo in the metadata area; but if --resource works why not.  The duplication of the logo was a nasty, temporary, workaround.

The epubs look particularly decent, but definitely lack in a some areas.  Doing lua filters, configs, ... etc is laborious though; maybe, for me at least, time would be better spent on producing content, or some programming, at least for now.  

Placing the pdfs (or pdf, or epub, or mobi, or dvi, or  manX, or *info, or ...) in the root was never intended. All output was originally intended as a direct subdir, e.g., ..../docs/contrib/[pdf || epub|| ...); but somehow they got  moved to root/pdf. name makepdf should definitely change.
So placing output in docs/contrib/[pdf || epub | ...]/content.[pdf || epub || ...] was actually a correct fix and the original intention.

I've no problem in placing them in root/[format]/contrib/content.format. I think the important thing is to ensure the tree structure is maintained and not everything  be placed in one directory. 
How to fix?
I can just keep everything as is and just create softlinks to root/[format] ; or simply dump them directly into root/[format]/ keeping the sub dir structure? It is really all the same to me.

Reply:
I think the best solution would be to add an option to place files in the root (not the default). Putting PDFs in the root is required for the CI because it needs to take this directory (which has only PDF files, no others) and tar it up as part of the release process. It could be an option specifically for use by the CI. Using `--resource` to fix the path for the logo seems to be the easiest solution to that problem.

Reply:
what _precisely_ do you mean by 'CI'?

Reply:
I've changed the script in a number of ways, generally in align with your comments:
Output is top-level-directory/[format]; where, per default, out is top-level-dir/pdf. If you desire epub, you must specify -t epub, and output is to top-level-dir/epub.
All output is silenced _except_ for errors. If you require additional information while running, then add the -v (verbose) to obtain additional information, which I've added.
It is a simple matter, well, relatively speaking,  to add additional formats by extending; just add the new format to the list of supported formats, which I placed near the top of the script, and add a method. One can easily use something other than pandoc, as it is all encapsulated in the one method. Output would then go to top-level-dir/my_new_format
The script has always exited on pandoc error. Try, for example, adding an incorrect argument to pandox, e.g., pandoc -whatever; and the script should exist, with pandoc error.

Reply:
If I use the current version, it doesn't abort on errors. This change however does and can be confirmed by running `./makepdf.sh -s opensolaris` which continues to go through all files in the current version (some have errors), in the version it stops after the first error.